classes/local/store/object_file_system.php

```diff
@@ -316,6 +321,50 @@ public function delete_object_from_local_by_hash($contenthash, $objectsize = 0)
         return $finallocation;
     }
 
+    /**
+     * @param string $rootpath
+     */
+    public function delete_empty_folders($rootpath) {
+        if (!is_dir($rootpath)) {
+            return;
+        }
+        if ($this->is_dir_empty($rootpath)) {
```

Comment:
This feels like is should be below the iterator below. ie delete all my empty children dirs first, and then I can delete myself